pkg/controller/extension/actuator_test.go

```diff
@@ -247,6 +250,52 @@ var _ = Describe("GenerateInstallManifest", func() {
 	})
 })
 
+var _ = Describe("ReconcileShootInfoConfigMap", func() {
+	var (
+		shootClient client.Client
+		config      *fluxv1alpha1.FluxConfig
+	)
+
+	It("should apply successfully and contain expected keys", func() {
+		shootName := "test-shoot"
+		technicalID := fmt.Sprintf("shoot--asdf-test-%s", shootName)
+		clusterIdentity := "magic-cluster-identity"
+		shootClient = newShootClient()
+		config = &fluxv1alpha1.FluxConfig{
+			Flux: &fluxv1alpha1.FluxInstallation{
+				Namespace: ptr.To("flux-system"),
+			},
+		}
+		cluster := &extensions.Cluster{
+			Shoot: &gardencorev1beta1.Shoot{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: shootName,
+				},
+				Status: gardencorev1beta1.ShootStatus{
+					TechnicalID:     technicalID,
+					ClusterIdentity: &clusterIdentity,
+				},
+			},
+		}
+		configMap := &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      shootInfoConfigMapName,
+				Namespace: *config.Flux.Namespace,
+			},
+		}
+
+		Expect(ReconcileShootInfoConfigMap(ctx, log, shootClient, config, cluster)).To(Succeed())
+
+		Expect(shootClient.Get(ctx, client.ObjectKeyFromObject(configMap), configMap))
+		Expect(len(configMap.Data)).To(Equal(3))
+		Expect(configMap.Data).To(Equal(map[string]string{
```

Comment:
Can you add another test case, where the configmap already exists? We should verify that it is correctly updated an unknown data keys are removed.